apps/api/src/app/content-templates/usecases/compile-template/email/compile-email-template.usecase.ts

```diff
@@ -0,0 +1,152 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { IEmailBlock, OrganizationRepository, OrganizationEntity } from '@novu/dal';
+import { CompileTemplate } from '../compile-template.usecase';
+import { CompileTemplateCommand } from '../compile-template.command';
+import { ApiException } from '../../../../shared/exceptions/api.exception';
+import * as fs from 'fs';
+import { CompileEmailTemplateCommand } from './compile-email-template.command';
+import { GetLayoutCommand, GetLayoutUseCase } from '../../../../layouts/use-cases';
+import { LayoutId } from '@novu/shared';
+
+const helperContent = fs.readFileSync(
+  `${__dirname}${
+    !process.env.E2E_RUNNER ? '/src/app/content-templates/usecases/compile-template' : '/../../compile-template'
+  }/templates/basic.handlebars`,
+  'utf8'
+);
+
+const defaultLayout = fs.readFileSync(
+  `${__dirname}${
+    !process.env.E2E_RUNNER ? '/src/app/content-templates/usecases/compile-template' : '/../../compile-template'
+  }/templates/layout.handlebars`,
+  'utf8'
+);
+
+@Injectable()
+export class CompileEmailTemplateUsecase {
+  constructor(
+    private compileTemplate: CompileTemplate,
+    private organizationRepository: OrganizationRepository,
+    private getLayoutUsecase: GetLayoutUseCase
+  ) {}
+
+  public async execute(command: CompileEmailTemplateCommand) {
+    const organization: OrganizationEntity | null = await this.organizationRepository.findById(command.organizationId);
+    if (!organization) throw new NotFoundException(`Organization ${command.organizationId} not found`);
+    let useNovuDefault = !command.layoutId;
+    let layout;
+
+    if (!useNovuDefault) {
+      try {
+        layout = await this.getLayoutUsecase.execute(
+          GetLayoutCommand.create({
+            layoutId: command.layoutId as LayoutId,
+            environmentId: command.environmentId,
+            organizationId: command.organizationId,
+          })
+        );
+      } catch (e) {
+        useNovuDefault = true;
+      }
+    }
+
+    const isEditorMode = command.contentType === 'editor';
+    let subject = '';
+    let content: string | IEmailBlock[] = command.content;
+    let preheader = command.preheader;
+    const layoutContent = useNovuDefault ? defaultLayout : layout.content;
```

Comment:
this variable should be defined before `if (!useNovuDefault)` with a default set to `defaultLayout`, so it will be a string type and you won't need to cast it to `string` later... like:
```
let layoutContent = defaultLayout;

if (!command.layoutId) {
...
layout = await this.getLayoutUsecase.execute(...
layoutContent = layout.content;
}

```
then you won't also need `as LayoutId` and `layoutContent as string`

Reply:
❤️